src/key_io.cpp

```diff
@@ -95,20 +95,26 @@ CTxDestination DecodeDestination(const std::string& str, const CChainParams& par
         error_str = "Invalid prefix for Base58-encoded address";
     }
     data.clear();
-    auto bech = bech32::Decode(str);
-    if (bech.second.size() > 0) {
+    const auto dec = bech32::Decode(str);
+    if ((dec.encoding == bech32::Encoding::BECH32 || dec.encoding == bech32::Encoding::BECH32M) && dec.data.size() > 0) {
```

Comment:
>> I honestly don't really like the std::tuple stuff. It's pretty ugly and a simple struct containing the 3 elements passed as a copy by reference (edit: I meant like the tuples are passed now) in the same way would be more straightforward. (@kallewoof)

> @kallewoof Agree, a struct is definitely warranted here. Updated. (@sipa)

No need to change this again, but in future you may find that structured bindings (since c++17) offer a clean and readable way to avoid the boilerplate of declaring a `struct` return. Combining with an if init statement:

```suggestion
    if (const auto [encoding, hrp, data] = bech32::Decode(str);
        (encoding == bech32::Encoding::BECH32 || encoding == bech32::Encoding::BECH32M) && data.size() > 0) {
        // [... do stuff ]
    } // encoding, hrp and data fall out of scope at end of if block
```

Reply:
Agree, though another reason why we may not want to do this: we'll want this PR backported as far as possible.

Reply:
Makes sense. I'll mark this as resolved.

Also, TIL: you can use structured bindings to unpack a `struct`, so this works even when the return value is a `DecodeResult`: `const auto [encoding, hrp, d] = bech32::Decode(str);`